### PR TITLE
Node module challenge

### PR DESCRIPTION
**Pull request by shoebpate1**

### CODE REVIEW

**Comment by J12934** [comment]
This looks really cool, very nice! 
I will try this out tomorrow 😀

File: routes/fileUpload.js [hunk 1]
```diff
@@ -2,15 +2,47 @@ const utils = require('../lib/utils')
...
+              .pipe(unzipper.Parse())
+              .on('entry', function (entry) {
+                var fileName = entry.path
+                var absolutePath = path.resolve('uploads/complaints/' + fileName)
+                if (absolutePath === path.resolve('ftp/legal.md') && utils.notSolved(challenges.fileWriteChallenge)) {
+                  utils.solve(challenges.fileWriteChallenge)
+                }
+                var juiceShopPath = path.resolve('.')
+                if (absolutePath.includes(juiceShopPath)) {
+                  entry.pipe(fs.createWriteStream('uploads/complaints/' + fileName).on('error', function (e) { console.log(e) }))
```
Comment (J12934) [inline routes/fileUpload.js:32]: This console.log gets trigger every time even when this is used in the regular not abusive way.

This is the error which gets logged:
```
{ Error: ENOENT: no such file or directory, open 'uploads/complaints/../../ftp/legal.md'
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'uploads/complaints/../../ftp/legal.md' }
```

Comment (J12934) [reply routes/fileUpload.js:32]: If this error is never relevant, i guess we should not log it.

**Review by J12934 — CHANGES_REQUESTED** [review]
Also i assumed that i would solve the upload type challenge by uploading the zip file.
But its not getting solved.

**Comment by J12934** [comment]
The challenge is really cool!
I created the zip archives with the tool you have send us and it worked like a charm 😉

File: routes/fileUpload.js (hunk 1 above)
Comment (shoebpate1) [reply routes/fileUpload.js:32]: Ohk so we just don't do anything with the error ?

Comment (J12934) [reply routes/fileUpload.js:32]: No wait, this error is actually valid.
The file was not really overwritten, but the challenge got solved

Comment (J12934) [reply routes/fileUpload.js:32]: Is the `legal.md` file properly overwritten on your machine @CaptainFreak ?

Comment (shoebpate1) [reply routes/fileUpload.js:32]: It works. Can you use the zip file i created which is in tests/files and check ?

Comment (J12934) [reply routes/fileUpload.js:32]: That also doesn't work on my machine

Comment (shoebpate1) [reply routes/fileUpload.js:32]: Actually it works smoothly for me. Always the above error is coming for you ?

Comment (shoebpate1) [reply routes/fileUpload.js:32]: Oh I forgot to tell .. The legal.md file in the archive is only differing in one character and that is an extra `s` in `##Terms of Use` . So u might have not been able to observe if any other error is not present.

Comment (shoebpate1) [reply routes/fileUpload.js:32]: It becomes `Uses`

Comment (J12934) [reply routes/fileUpload.js:32]: Mh no that also doesnt change.
I tried running it in docker and it also doesn't appear to be working in docker.
Can you build the container locally and test it with docker on your machine?

Comment (shoebpate1) [reply routes/fileUpload.js:32]: Oh am not that familiar with docker. So can I make container of my own fork ?

Comment (shoebpate1) [reply routes/fileUpload.js:32]: What errors you are getting ?

Comment (J12934) [reply routes/fileUpload.js:32]: It the same error as above

Comment (J12934) [reply routes/fileUpload.js:32]: You can build and start the docker container like this (assuming you have docker installed):
```sh
docker build -t juice-shop-zip .
docker run -p 3000:3000 juice-shop-zip
```

Comment (shoebpate1) [reply routes/fileUpload.js:32]: Oh got it. So in my local branch I do have the required folders ```uploads/complaints/``` but I did not knew that empty directories dont get commited !

Comment (shoebpate1) [reply routes/fileUpload.js:32]: So should we just add .gitignores to the directories ,So that they get committed ?

Comment (J12934) [reply routes/fileUpload.js:32]: Personally prefer .gitkeep but not .gitignore would also work.
See: https://stackoverflow.com/questions/7229885/what-are-the-differences-between-gitignore-and-gitkeep

Comment (shoebpate1) [reply routes/fileUpload.js:32]: Yea chose .gitkeep already :)

Comment (J12934) [reply routes/fileUpload.js:32]: Okay very nice its now working properly!

**Review by J12934 — APPROVED** [review]

**Pull request merged**